flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskExecutorTest.java

```diff
@@ -1472,22 +1349,12 @@ public void testIgnoringSlotRequestsIfNotRegistered() throws Exception {
 	 */
 	@Test
 	public void testReconnectionAttemptIfExplicitlyDisconnected() throws Exception {
-		final long heartbeatInterval = 1000L;
 		final TaskSlotTable taskSlotTable = new TaskSlotTable(Collections.singleton(ResourceProfile.UNKNOWN), timerService);
 		final TaskManagerLocation taskManagerLocation = new LocalTaskManagerLocation();
-		final TaskExecutor taskExecutor = new TaskExecutor(
-			rpc,
-			TaskManagerConfiguration.fromConfiguration(configuration),
-			haServices,
-			new TaskManagerServicesBuilder()
-				.setTaskSlotTable(taskSlotTable)
-				.setTaskManagerLocation(taskManagerLocation)
-				.build(),
-			new HeartbeatServices(heartbeatInterval, 1000L),
-			UnregisteredMetricGroups.createUnregisteredTaskManagerMetricGroup(),
-			null,
-			dummyBlobCacheService,
-			testingFatalErrorHandler);
+		final TaskExecutor taskExecutor = createTaskExecutor(new TaskManagerServicesBuilder()
```

Comment:
Why are we no longer passing `new HeartbeatServices(heartbeatInterval, 1000L)`?

Reply:
do you think we need always a new one? cannot we share `HEARTBEAT_SERVICES = new HeartbeatServices(1000L, 1000L);` as it is now in the PR?

Reply:
I thought `HeartbeatServices`  could be reused in the tests only if it needs different internal and timeout in special tests. In this test the created new `HeartbeatServices` is the same with `HEARTBEAT_SERVICES`, so I reuse the static one. Referring some other previous tests, they also reuse the same service.  Do you have other concerns? @zentol

Reply:
I don't know whether we need a separate heartbeat service. My general approach to tests is to not change such details, as there could be repercussions one isn't aware of.

I'm not really concerned about correctness in this case; my point essentially boils down to "Had you not changed this detail we wouldn't even have to worry about these changes potentially affecting the test."

Reply:
Thanks for the explanation. 

Ok, that makes sense to keep the same behavior during refactoring. Even though it is not necessary to create a new heartbeat service as we thought, it should be in a separate ticket for this issue, not involved in my changes.